### PR TITLE
Add graph & api call to worker2

### PR DESCRIPTION
**Pull request by jennyf19**
https://github.com/AzureAD/microsoft-identity-web/issues/900

Still need to test and build the template from the branch.

### CODE REVIEW

File: ProjectTemplates/templates/Worker-CSharp/Company.Application1.csproj [hunk 1]
```diff
@@ -13,7 +13,9 @@
 
   <ItemGroup>
     <PackageReference Include="Grpc.AspNetCore" Version="2.32.0" />
-    <!--<PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="5.0.0-0" />-->
     <PackageReference Include="Microsoft.Identity.Web" Version="1.*" />
+    <!--#if (GenerateGraph) -->
```
Comment (bgavrilMS) [inline ProjectTemplates/templates/Worker-CSharp/Company.Application1.csproj:17]: Huh, I didn't know about this trick. How does it work?

File: ProjectTemplates/templates/Worker-CSharp/Startup.cs [hunk 2]
```diff
@@ -28,15 +31,31 @@ public void ConfigureServices(IServiceCollection services)
 
 #if (OrganizationalAuth)
             services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+#if (GenerateApiOrGraph)
```
Comment (bgavrilMS) [inline ProjectTemplates/templates/Worker-CSharp/Startup.cs:34]: Could this be structured without nesting all these #if clauses? smth like..

#if A
// code
#endif 
#if B
// code
#end B

File: ProjectTemplates/templates/Worker-CSharp/Startup.cs [hunk 3]
```diff
@@ -28,15 +31,31 @@ public void ConfigureServices(IServiceCollection services)
...
+                .AddMicrosoftIdentityWebApi(Configuration.GetSection("AzureAd"))
+                    .EnableTokenAcquisitionToCallDownstreamApi()
```
Comment (bgavrilMS) [inline ProjectTemplates/templates/Worker-CSharp/Startup.cs:36]: nit: align left

**Review by bgavrilMS — APPROVED** [review]

**Review by pmaytak — APPROVED** [review]

File: ProjectTemplates/templates/Worker-CSharp/Company.Application1.csproj (hunk 1 above)
Comment (jennyf19) [reply ProjectTemplates/templates/Worker-CSharp/Company.Application1.csproj:17]: the dotnet new tool will do the template extensions based on the command line options and the variables computed there. `GenerateGraph` is one of those variables, it's defined [here](https://github.com/AzureAD/microsoft-identity-web/pull/901/files#diff-70d6aa29757c7ed2d2380a8ba03f394e95b5585f3e4604b48c74625b592147e3R191)

File: ProjectTemplates/templates/Worker-CSharp/Startup.cs (hunk 2 above)
Comment (jennyf19) [reply ProjectTemplates/templates/Worker-CSharp/Startup.cs:34]: we try to remain as close as possible to the asp .net core templates. they are very picky w/their templates.

File: ProjectTemplates/templates/Worker-CSharp/Startup.cs (hunk 3 above)
Comment (jennyf19) [reply ProjectTemplates/templates/Worker-CSharp/Startup.cs:36]: it's a subsection.

File: ProjectTemplates/templates/Worker-CSharp/GreeterService.cs [hunk 4]
```diff
@@ -13,26 +23,87 @@ namespace Company.Application1
...
+                throw new HttpRequestException($"Invalid status code in the HttpResponseMessage: {response.StatusCode}: {error}");
+            }
+
+            return Task.FromResult(new HelloReply
+            {
+                Message = "Hello " + request.Name
+            });
+        }
+
+# elseif (GenerateGraph)
```
Comment (jmprieur) [inline ProjectTemplates/templates/Worker-CSharp/GreeterService.cs:65]: There is an extra space between `#` and `elseif`.

**Review by jmprieur — CHANGES_REQUESTED** [review]
This is an awesome start @jennyf19 !
I realize the amount of work you must have put to get to this outcome, considering how complex the templates are, and how hard and iterative it is to test them

There is little typo in the processor directive in `GreeterService.cs` 
This means the template for this file is not correctly expanded when running the following command:

```Shell
dotnet new worker2 --auth SingleOrg --called-api-url "https://graph.microsoft.com/beta/me" --called-api-scopes "user.read"
```

File: ProjectTemplates/templates/Worker-CSharp/GreeterService.cs (hunk 4 above)
Comment (jennyf19) [reply ProjectTemplates/templates/Worker-CSharp/GreeterService.cs]: yes, nice catch. thanks.

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @jennyf19 

I tests all the cases, and it expands and builds all. Awesome!

**Pull request merged**